filebeat/harvester/reader/multiline.go

```diff
@@ -260,8 +282,14 @@ func (mlr *Multiline) addLine(m Message) {
 	mlr.readBytes += m.Bytes
 }
 
-func (mlr *Multiline) reset() {
-	mlr.state = (*Multiline).readFirst
+// reset sets state of the reader to readFirst
+func (mlr *Multiline) resetState() {
+	mlr.setState((*Multiline).readFirst)
+}
+
+// setState sets state to the given function
+func (mlr *Multiline) setState(next func(mlr *Multiline) (Message, error)) {
```

Comment:
Should we go with 3 variables?
```
readFirst
readNext
readFailed
```

We still have the ugly syntax but somehow hidden